.github/workflows/shellcheck.yml

```diff
@@ -0,0 +1,11 @@
+name: Run ShellCheck on PR
```

Comment:
We should have this GHA in [ Code Hygiene ](https://github.com/opensearch-project/OpenSearch/blob/main/.github/workflows/code-hygiene.yml). @reta @laysauchoa WDYT?

Reply:
@owaiskazi19 I think it may fit in there, yeah

Reply:
done!